frontend/src/metabase-lib/filter.ts

```diff
@@ -50,3 +51,11 @@ export function filterOperator(
 ) {
   return ML.filter_operator(query, stageIndex, filterClause);
 }
+
+export function filterParts(
+  query: Query,
+  stageIndex: number,
+  filterClause: FilterClause,
+): FilterParts {
+  return ML.filter_parts(query, stageIndex, filterClause);
+}
```

Comment:
I like the idea of making `filter_parts` a separate method 👍

Reply:
@metamben curios if there's a reason why `externalOp` only needed a `FilterClause`, but `filterParts` also needs `query` and `stageIndex`?

Reply:
They are used to produce `operator`. In fact, this function is a generalization of `filterOperator`. We might consider removing that.